src/components/wallet/TokenAmount.js

```diff
@@ -3,9 +3,12 @@ import { formatTokenAmount } from '../../utils/amounts'
 
 const FRAC_DIGITS = 5
 
-const TokenAmount = ({ token: { balance, decimals, symbol}, className }) => (
+const TokenAmount = ({ token: { balance, decimals, symbol }, className }) => (
     <div className={className} title={showFullAmount(balance, decimals, symbol)}>
-        {balance && formatToken(balance, decimals)}
+        {balance
+            ? formatToken(balance, decimals)
+            : <span className='dots' />
```

Comment:
it is probably best not to spin dots for every balance being loaded (screen gonna look too busy, especially given that balances load pretty fast)

I think if we are to improve something in loading is actually try to make not loaded state closer to loaded (e.g. right now one item is smaller in height before loaded, so it kinda looks glitchy)

Reply:
> I think if we are to improve something in loading is actually try to make not loaded state closer to loaded (e.g. right now one item is smaller in height before loaded, so it kinda looks glitchy)

This is fixed, by blocking showing token until metadata are available

I'm also not sure about the dots that appear in the balance place until the balance is loaded. But I wanted to handle situations when the network will be slow. I was thinking about another solution when we don't display token at all until the balance is loaded. In this case, dots will be visible only on section title (Tokens...)
What do you think?